src/esp/physics/bullet/BulletDebugManager.cpp

```diff
@@ -114,5 +132,55 @@ std::string BulletDebugManager::getStepCollisionSummary(
   return s.str();
 }
 
+std::string BulletDebugManager::getCollisionFilteringSummary(bool doVerbose) {
+  std::stringstream s2;
+  for (const auto& pair0 : collisionObjectToDebugName_) {
+    const btCollisionObject* colObj0 = pair0.first;
+    s2 << getDebugStringForCollisionObject(colObj0) << std::endl;
+  }
+
+  if (doVerbose) {
+    for (const auto& pair0 : collisionObjectToDebugName_) {
+      std::stringstream s;
+      const btCollisionObject* colObj0 = pair0.first;
+      s << getDebugStringForCollisionObject(colObj0)
+        << " may collide with:" << std::endl;
+
+      bool foundAny = false;
+      for (const auto& pair1 : collisionObjectToDebugName_) {
+        const btCollisionObject* colObj1 = pair1.first;
+        const std::string& name1 = pair1.second;
+        if (colObj1 == colObj0) {
+          continue;
+        }
+
+        const auto* broadphaseHandle0 = colObj0->getBroadphaseHandle();
+        const auto* broadphaseHandle1 = colObj1->getBroadphaseHandle();
+
+        // logic copied from
+        // btHashedOverlappingPairCache::needsBroadphaseCollision
+        bool collides = (broadphaseHandle0->m_collisionFilterGroup &
+                         broadphaseHandle1->m_collisionFilterMask) != 0;
+        collides = collides && (broadphaseHandle1->m_collisionFilterGroup &
+                                broadphaseHandle0->m_collisionFilterMask);
+
+        if (collides) {
+          s << "  " << name1 << std::endl;
+          foundAny = true;
+        }
+      }
+
+      if (!foundAny) {
+        s << "  (none)" << std::endl;
+      }
+
+      LOG(INFO) << s.str();
```

Comment:
This is pretty sloppy. Conceptually, all this verbose output should just be appended to s2 and returned as part of the return-value string. Instead, I'm logging directly here, because I hit an issue where printing a very long string to stdout wasn't working very well with VS Code's terminal.